pkg/planner/cardinality/selectivity.go

```diff
@@ -1189,14 +1189,21 @@ func outOfRangeFullNDV(ndv, origRowCount, notNullCount, realtimeRowCount, increa
 	if newRows < 0 {
 		newRows = min(notNullCount, realtimeRowCount)
 	}
-	// if no NDV - derive an NDV using sqrt
+	// if no NDV - derive an NDV using sqrt, this could happen for unanalyzed tables
 	if ndv <= 0 {
 		ndv = math.Sqrt(max(notNullCount, realtimeRowCount))
 	} else {
 		// We need to increase the ndv by increaseFactor because the estimate will be increased by
 		// the caller of the function
 		ndv *= increaseFactor
 	}
+	// If topN represents all NDV values, the NDV should be relatively small.
+	// Small NDV could cause extremely inaccurate result, use `outOfRangeBetweenRate` to smooth the result.
+	// For example, TopN = {(value:1, rows: 10000), (2, 10000), (3, 10000)} and newRows = 15000, we should assume most
+	// newly added rows are 1, 2 or 3. Then for an out-of-range estimation like `where col=9999`, the result should be
+	// close to 0, but if we still use the original NDV, the result could be extremely large: 15000/3 = 5000.
+	// See #64137 for a concrete example.
+	ndv = max(ndv, float64(outOfRangeBetweenRate)) // avoid inaccurate estimate caused by small NDV
```

Comment:
so the issue case ndv is max(2, 100) = and get 100 here

Reply:
Yes, and actually this is what we did before, here we just keep the same behavior: use `outOfRangeBetweenRate` as the lower-bound of NDV for out-of-range estimation
<img width="828" height="632" alt="image" src="https://github.com/user-attachments/assets/606acf3a-15d6-4b82-b917-5365c80e36ba" />